Add unleash edge quickstart guide (#10809)
Unleash/unleash: website/docs/how-to/unleash-edge-quickstart.md

---

> title: Unleash Edge Quickstart Guide

Reviewer: ```suggestion
title: Unleash Edge quickstart
```

slug: /how-to/unleash-edge-quickstart
---

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

This document helps you get started with [Unleash Edge](https://docs.getunleash.io/reference/unleash-edge) locally.

## Why Unleash Edge

> Unleash Edge as a lightweight layer between your SDKs and your Unleash instance.

Reviewer: ```suggestion
Unleash Edge is a lightweight layer between your SDKs and your Unleash instance.
```

> It exposes the same HTTP interface while providing higher throughput.

Reviewer: ```suggestion
It exposes the same HTTP interface as the main [Unleash API](/api-overview) but is built for higher throughput and lower latency.
```

> 

Reviewer: I'm missing a bridging sentence here. Something like:
Before we dive into the overall setup, let’s look at the specific performance and resilience benefits you get from running Edge.

    Optionally, you can set up multiple layers between your application and the Unleash instance (daisy-chaining). This lets you define the optimal configuration for caching and resilience, while adapting Unleash Edge to your architecture's topology.
  </TabItem>

  <TabItem value="offline" label="Offline mode">
    Unleash Edge can also run in [offline mode](https://github.com/Unleash/unleash-edge/tree/main/examples), so it doesn't need a connection to an upstream Unleash instance. This simplifies local development and helps in environments with limited connectivity.
  </TabItem>

</Tabs>

## Prerequisites

Here's what you need before getting started:

1. An [Unleash instance](https://docs.getunleash.io/quickstart) running locally or remotely (version `4.15` or later)
2. A valid [API token](https://docs.getunleash.io/reference/api-tokens-and-client-keys) for your Unleash instance
3. [Docker](https://www.docker.com/get-started/) installed and running
4. Your preferred Unleash SDK in a [sample app](https://github.com/Unleash/unleash-sdk-examples)

> ## How to run Unleash Edge locally

Reviewer: For task-based stuff, we typically go for plain form, not question form https://developers.google.com/style/headings

But I'm happy to experiment with it here

  <TabItem value="docker" label="Docker">
    Launch Unleash Edge locally with Docker:

    ```shell
    docker run -it \
      -p 3063:3063 \
      -e STRICT=true \
      -e UPSTREAM_URL=<your_unleash_instance> \
      -e TOKENS='<your_backend_token>' \
      unleashorg/unleash-edge \
      edge
    ```

    :::info[Docker Image versioning]

    By default, the command above uses the latest `unleashorg/unleash-edge` tag.

    You can also pick a specific version from the list of [tags on Docker Hub](https://hub.docker.com/r/unleashorg/unleash-edge).

    :::

  </TabItem>

  <TabItem value="docker-compose" label="Docker Compose">
    Launch the [examples/docker-compose.yml](https://github.com/Unleash/unleash-edge/blob/main/examples/docker-compose.yml) file:

    ```shell
    git clone https://github.com/unleash/unleash-edge/
    cd unleash-edge/examples
    docker compose up
    ```

    :::info[Docker Compose services]

    Please note that the sample Docker Compose file includes a Redis service for caching.
    In case you want to run Unleash Edge without Redis, make sure to remove the `REDIS_URL` environment variable and the Redis service.

    :::

  </TabItem>
</Tabs>

> #### Required parameters

Reviewer: ```suggestion
### Required parameters
```

Let's break down the parameters you need to replace in the launch command.

> ### `<your_unleash_instance>`

Reviewer: I'd vote against code formatting in headings, unless necessary. If you want to keep them, then we need to fix the hierarchy, so these should be l4

https://developers.google.com/style/headings#heading-and-title-format

Author: Updated 👌

    Note: make sure you keep the single quotes in `--tokens '...'` so the `*` isn't expanded by your shell.
  </TabItem>

  <TabItem value="docker" label="Docker">
    This API token is required in strict mode, which is recommended since `v19.2`.

    You can generate a new API token by visiting your Unleash instance, under **Admin settings > Access control > API access**. Click **New API token**, give it a name, and confirm the default values.

    Note: make sure you keep the single quotes in `-e TOKENS='...'` so the `*` isn't expanded by your shell.
  </TabItem>

  <TabItem value="docker-compose" label="Docker Compose">
    This parameter is managed for you in the Docker Compose file.

    You can customize the default tokens by editing the Unleash environment variables:

    ```yaml
    services:
      unleash:
        environment:
          INIT_FRONTEND_API_TOKENS: "default:development.unleash-insecure-frontend-api-token"
          INIT_CLIENT_API_TOKENS: "default:development.unleash-insecure-api-token"
    ```

>     Unleash Edge is automatically configured with the same (backend) token.

Reviewer: ```suggestion
    Unleash Edge is automatically configured with the same backend token.
```
